nemo_curator/modules/exact_dedup.py

```diff
@@ -135,7 +136,7 @@ def hash_documents(
             # TODO: Generalize ty using self.hash_method
             return df.apply(lambda x: md5(x.encode()).hexdigest())
 
-    def __call__(self, dataset: DocumentDataset) -> Union[DocumentDataset, str]:
+    def identify(self, dataset: DocumentDataset) -> DocumentDataset:
```

Comment:
```suggestion
    def _identify(self, dataset: DocumentDataset) -> DocumentDataset:
```
Nit, but maybe call them `_identify` and `_remove` if they are not intended to be accessed by the user directly.

Reply:
Let's keep them exposed especially since remove won't work at scales where size of duplicate >> host memory, in which case the user will need to break down identify and remove

Reply:
Yes, that makes sense to me. What about calling it `identify_duplicates`?

Reply:
Feels more verbose than necessary `XYZDuplicates.identify_duplicates(..)` vs `XYZDuplicates.identify`.

Not strongly opinionated tho, can change it. One argument in favor of identify_duplicates would be that in future we might want to expose `identify_documents_to_keep` in which the distinction might be necessary

 Let's just get one more persons thought

cc @ayushdg / @ryantwolf / @VibhuJawa

Reply:
Sounds good, initially I thought it's slightly verbose, but another argument in favor of `identify_duplicates` would be that in future we might want to expose `identify_documents_to_keep` in which the distinction might be necessary


cc @ayushdg / @ryantwolf / @VibhuJawa